arviz/plots/draw_prior.py

```diff
@@ -0,0 +1,251 @@
+"""Prior elicitation using roulette mode."""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.stats as stats
+import matplotlib.patches as patches
+from matplotlib.widgets import RadioButtons, TextBox
+from math import ceil, floor
+
+from ..rcparams import rcParams
+
+
+def create_grid(x_min=0, x_max=1, nrows=10, ncols=10, ax=None):
+    xx = np.arange(0, ncols)
+    yy = np.arange(0, nrows)
+
+    if ncols < 10:
+        num = ncols
+    else:
+        num = 10
+
+    ax[0, 0].set(
+        xticks=np.linspace(0, ncols, num=num),
+        xticklabels=[f"{i:.1f}" for i in np.linspace(x_min, x_max, num=num)]
+        ,
+    )
+
+    coll = np.zeros((nrows, ncols), dtype=object)
+    for idx, xi in enumerate(xx):
+        for idy, yi in enumerate(yy):
+            sq = patches.Rectangle((xi, yi), 1, 1, fill=True, facecolor="0.8", edgecolor="w")
+            ax[0, 0].add_patch(sq)
+            coll[idy, idx] = sq
+    return coll
+
+
+class Rectangles:
+    def __init__(self, fig, coll, nrows, ncols, ax):
+        self.fig = fig
+        self.coll = coll
+        self.nrows = nrows
+        self.ncols = ncols
+        self.ax = ax
+        self.weights = {k: 0 for k in range(0, ncols)}
+        fig.canvas.mpl_connect("button_press_event", self)
+
+    def __call__(self, event):
+        if event.inaxes == self.ax[0, 0]:
+            x = event.xdata
+            y = event.ydata
+            idx = floor(x)
+            idy = ceil(y)
+            if self.weights[idx] >= idy:
+                idy -= 1
+                for row in range(self.nrows):
+                    self.coll[row, idx].set_facecolor("0.8")
+            self.weights[idx] = idy
+            for idy in range(idy):
+                self.coll[idy, idx].set_facecolor("C1")
+            self.fig.canvas.draw()
+
+
+def on_leave_fig(event, grid, radio, lower_q, upper_q, min_range, max_range, ncols, ax):
+    x_min = float(min_range.text)
+    x_max = float(max_range.text)
+    x_range = x_max - x_min
+    samples = []
+    ok = 0
+
+    if any(grid.weights.values()):
+        for k, v in grid.weights.items():
+            if v != 0:
+                ok += 1
+            la = np.repeat((k / ncols * x_range) + x_min + ((x_range / ncols) / 2), v*100+1)
+            samples.extend(la)
+
+        x = np.linspace(x_min, x_max, 500)
+        dist_name = radio.value_selected
+        if dist_name == "automatic":
+            sum_pdf_old = -np.inf
+            for dist in [stats.beta, stats.norm, stats.gamma, stats.lognorm]:
+                if dist.name == "beta":
+                    a, b, _, _ = dist.fit(samples, floc=x_min, fscale=x_range)
```

Comment:
this should work with just `dist_args = dist.fit(samples, ...)` and `can_dist = dist(*args)`. I guess you're restricting the distribution to have support in `(x_min, x_max)`, but then the fit could be arbitrarily bad, I think!

Reply:
This needs a little bit more care so we can match scipy parameterization with PyMC (and other ppls parameterization). The fit seems to work well in practice. I compared with results from the MATCH site and with scipy, but I admit that these tests were not exhaustive.